server/server.go

```diff
@@ -0,0 +1,83 @@
+package main
+
+import (
+	"log"
+	"net"
+	"net/rpc"
+	"sync"
+
+	"github.com/lukedmor/conduit_tiny/common"
+)
+
+type Provider struct {
+	cs *conduitServer
+}
+
+func (p *Provider) Join(a *common.ProviderJoinLeaveArgs, reply *common.Nothing) error {
+	p.cs.providers[a.Addr] = true
+	return nil
+}
+
+func (p *Provider) Leave(a *common.ProviderJoinLeaveArgs, reply *common.Nothing) error {
+	delete(p.cs.providers, a.Addr)
```

Comment:
What if a provider crashes before calling Leave? There should be another way that a provider may get removed. Perhaps if a client failed to reach the provider, the client could let the server know (and the server could maybe verify the truth to this)